pkg/query-service/app/querier/v2/querier.go

```diff
@@ -345,15 +346,27 @@ func (q *querier) runLogsListQuery(ctx context.Context, params *v3.QueryRangePar
 
 		// append a filter to the params
 		if len(data) > 0 {
-			params.CompositeQuery.BuilderQueries[qName].Filters.Items = append(params.CompositeQuery.BuilderQueries[qName].Filters.Items, v3.FilterItem{
-				Key: v3.AttributeKey{
-					Key:      "id",
-					IsColumn: true,
-					DataType: "string",
-				},
-				Operator: v3.FilterOperatorLessThan,
-				Value:    data[len(data)-1].Data["id"],
-			})
+			if params.CompositeQuery.BuilderQueries[qName].DataSource == v3.DataSourceLogs {
+				params.CompositeQuery.BuilderQueries[qName].Filters.Items = append(params.CompositeQuery.BuilderQueries[qName].Filters.Items, v3.FilterItem{
+					Key: v3.AttributeKey{
+						Key:      "id",
+						IsColumn: true,
+						DataType: "string",
+					},
+					Operator: v3.FilterOperatorLessThan,
+					Value:    data[len(data)-1].Data["id"],
+				})
+			} else {
+				// for traces setting offset = 0 works
+				// eg -
+				// 1)--- searching 100 logs in between t1, t10, t100 with offset 0
+				// if 100 logs are there in t1 to t10 then 100 will return immediately.
+				// if 10 logs are there in t1 to t10 then we get 10, set offset to 0 and search in the next timerange of t10 to t100.
+				// 1)--- searching 100 logs in between t1, t10, t100 with offset 100
+				// It will have offset = 0 till 100 logs are found in one of the timerange tx to tx-1
+				// If it finds <100 in tx to tx-1 then it will set offset = 0 and search in the next timerange of tx-1 to tx-2
+				params.CompositeQuery.BuilderQueries[qName].Offset = 0
```

Comment:
Why is offset hard coded to 0? When frontend makes a request `offset=100,limit=50` what happens?

Reply:
> it will have to set offset=0, and limit=20

Still not going to be correct. Please take a couple of examples of the client moving pages and show how it is going to return the correct result.

Reply:
My bad it should be been, offset=0, and limit=30.

I have updated the example with proper code and example.

```
// 1)offset = 0, limit = 100, tsRanges = [t1, t10], [t10, 20], [t20, t30]
//
// if 100 traces are there in [t1, t10] then 100 will return immediately.
// if 10 traces are there in [t1, t10] then we get 10, set offset to 0 and limit to 90, search in the next timerange of [t10, t20]
// if we don't find any trace in [t1, t10], then we search in [t10, t20] with offset=0, limit=100

//
// 2) offset = 50, limit = 100, tsRanges = [t1, t10], [t10, r20], [t20, t30]
//
// If we find 100 traces in [t1, t10] then we return immediately
// If we find 50 in [t1, t10] then it will set offset = 0 and limit = 50 and search in the next timerange of [t10, r20]
// if we don't find any trace in [t1, t10], then we search in [t10, r20] with offset=50, limit=100
			
```